sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -179,6 +179,7 @@ private boolean shouldApplyScopeData(
           buildEnvelope(event, attachments, session, traceContext, null);
 
       if (envelope != null) {
+        hint.clear();
```

Comment:
I think we can still clear it, even if the envelope is null, so probably just move it before `if`